### ISSUE TITLE
Intermittent TIMEOUT in /2dcontext/imagebitmap/canvas-createImageBitmap-video-resize.html

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#22010]
```
  ▶ TIMEOUT [expected OK] /2dcontext/imagebitmap/canvas-createImageBitmap-video-resize.html
  │ 
  │ VMware, Inc.
  │ softpipe
  │ 3.3 (Core Profile) Mesa 18.1.0-devel
  └ Setup media player error SetPropertyFailed("Failed to set config")

  ▶ Unexpected subtest result in /2dcontext/imagebitmap/canvas-createImageBitmap-video-resize.html:
  └ NOTRUN [expected FAIL] createImageBitmap(HTMLVideoElement) with resize option
```

**Comment by ferjm** [comment]
This error likely comes from [here](https://github.com/servo/media/blob/master/backends/gstreamer/src/player.rs#L250). I am not sure why it would fail setting the config though. We saw this in the past and [temporarily removed setting the config](https://github.com/servo/media/pull/144) because we were not doing anything with the position, but now we need this.

@ceyusa @sdroege any idea why it may fail setting the config?

**Comment by ferjm** [comment]
I found this on the log:

```
gst-player gstplayer.c:4500:gst_player_set_config:<player1> can't change config while player is buffering
```

**Comment by sdroege** [comment]
How would that timeout if setting the config failed? That should just be returning an error.

You can only set the player config before the player is started, or after it was stopped. Do it right after construction in your case, I guess

**Comment by ferjm** [comment]
Apparently, we need to be in the `GST_PLAYER_STATE_STOPPED` state to be able to set the config:

https://cgit.freedesktop.org/gstreamer/gst-plugins-bad/tree/gst-libs/gst/player/gstplayer.c#n4498

However, we don't need to set the entire config. We just need to set the position update interval. And luckily the method to set this config does not have the state constraint:

https://cgit.freedesktop.org/gstreamer/gst-plugins-bad/tree/gst-libs/gst/player/gstplayer.c#n4585

[gstreamer-rs has bindings](https://github.com/sdroege/gstreamer-rs/blob/a3dffc2faa1981111f8b80b491d567470ec2f812/gstreamer-player/src/config.rs#L73) for this method and we are already using it in servo-media [here](https://github.com/servo/media/blob/master/backends/gstreamer/src/player.rs#L247). But we are also trying to set the entire config [immediately after that](https://github.com/servo/media/blob/master/backends/gstreamer/src/player.rs#L248), which does not seem to be necessary and it's likely the source of this error.

**Comment by ferjm** [comment]
> How would that timeout if setting the config failed? That should just be returning an error.

The individual tests are returning an error, but the testharness timeouts. Well, it used to timeout before #20755 was merged. Now it seems to always error as well if the setup fails.

**Comment by sdroege** [comment]
That function is setting the interval on the config, you still need to set the config on the player and that only works in stopped state. Why do you need to be able to change the interval after the player is started? Maybe there's something we can improve here on the API

**Comment by ferjm** [comment]
> Why do you need to be able to change the interval after the player is started?

We don't :)

IIUC what we are trying to do is to change the interval **before** the player is started, but we are apparently failing at that.

We set the config [here](https://github.com/servo/media/blob/master/backends/gstreamer/src/player.rs#L248) and player construction happens [here](https://github.com/servo/media/blob/master/backends/gstreamer/src/player.rs#L237). The only thing that happens in between is [setting the player's uri](https://github.com/servo/media/blob/master/backends/gstreamer/src/player.rs#L241), does that start the player?

**Comment by ferjm** [comment]
https://gist.github.com/ferjm/d961fabdd03a43474133f00cd5f1db96

**Comment by sdroege** [comment]
Checking, there's a bug somewhere and it's not in your code :)

As a workaround I'd recommend setting the URI after the other setup is done for now

**Comment by sdroege** [comment]
Found the culprit gst_player_stop_internal() sets the state to `READY` and it's called by `gst_player_set_uri_internal()`. Should check in there first if we should actually go to `READY` or not. We shouldn't really unless the current state is >= `READY`.

I'll fix in a bit but that will need GStreamer 1.14.5/1.15.1 then (so work around it like I mentioned above)

**Comment by ferjm** [comment]
Ok, thanks for your help!

**Comment by sdroege** [comment]
No problem, sorry for the bug :) I guess so far everybody did configuration before setting the uri and that's why it was not noticed

**Comment by sdroege** [comment]
http://cgit.freedesktop.org/gstreamer/gst-plugins-bad/commit/?id=afbbc3a97ec391c6a582f3c746965fdc3eb3e1f3

Fix will be in 1.14.5 / 1.15.1
